---
title: C28205
ms.date: 11/04/2016
ms.topic: reference
f1_keywords:
- C28205
helpviewer_keywords:
- C28205
ms.assetid: 3d802885-bdb8-47cb-9108-527a328a0774
ms.openlocfilehash: 18c97b7185628e8b736651dc0ea902216e23b769
ms.sourcegitcommit: 7bea0420d0e476287641edeb33a9d5689a98cb98
ms.translationtype: MT
ms.contentlocale: it-IT
ms.lasthandoff: 02/17/2020
ms.locfileid: "77416464"
---
# <a name="c28205"></a>C28205

> avviso C28205: funzione >: `_Success_` o `_On_failure_` utilizzato in un contesto non valido

Le annotazioni `_Success_` e `_On_failure_` possono essere utilizzate solo sui valori restituiti dalla funzione.

## <a name="examples"></a>Esempi

```cpp
#include<sal.h>

// Oops, _Success_ is not valid in parameter lists, should be moved to return value.
bool GetValue( _Success_(return != false) _Out_ int *pInt, bool flag)
{
   if(flag) {
      *pInt = 5;
      return true;
   } else {
      return false;
   }
}
```

```cpp
#include<sal.h>

_Success_(return != false)
bool GetValue(_Out_ int *pInt, bool flag)
{
   if(flag) {
      *pInt = 5;
      return true;
   } else {
      return false;
   }
}
```

[Informazioni su Sal](understanding-sal.md)
